### ISSUE TITLE
Remove not exist cache key got exception related to Resource Not Found (404)

### ISSUE THREAD

**Issue by thohng** [Azure/Microsoft.Extensions.Caching.Cosmos#37]
Hi,

We got exception `Microsoft.Azure.Cosmos.CosmosException : Response status code does not indicate success: NotFound (404)` when remove a not exist cache key.

```
Microsoft.Azure.Cosmos.CosmosException : Response status code does not indicate success: NotFound (404); Substatus: 0; ActivityId: 970d262f-dfcf-4b01-8ec1-1a1f25ee18bb; Reason: ({
  "Errors": [
    "Resource Not Found. Learn more: https://aka.ms/cosmosdb-tsg-not-found"
  ]
});
   at Microsoft.Azure.Cosmos.ResponseMessage.EnsureSuccessStatusCode()
   at Microsoft.Azure.Cosmos.CosmosResponseFactoryCore.ProcessMessage[T](ResponseMessage responseMessage, Func`2 createResponse)
   at Microsoft.Azure.Cosmos.CosmosResponseFactoryCore.CreateItemResponse[T](ResponseMessage responseMessage)
   at Microsoft.Azure.Cosmos.ContainerCore.DeleteItemAsync[T](CosmosDiagnosticsContext diagnosticsContext, String id, PartitionKey partitionKey, ItemRequestOptions requestOptions, CancellationToken cancellationToken)
   at Microsoft.Azure.Cosmos.ClientContextCore.RunWithDiagnosticsHelperAsync[TResult](CosmosDiagnosticsContext diagnosticsContext, Func`2 task)
   at Microsoft.Extensions.Caching.Cosmos.CosmosCache.RemoveAsync(String key, CancellationToken token)
....
```

**Comment by marcosikkens** [comment]
Hi i've experienced the same problem.  Lets hope it will be merged soon.

**Comment by ealsur** [comment]
Fix released in 1.0.0-preview6: https://www.nuget.org/packages/Microsoft.Extensions.Caching.Cosmos/1.0.0-preview6

Changelog updated: https://github.com/Azure/Microsoft.Extensions.Caching.Cosmos/blob/master/changelog.md#-100-preview6---2021-03-08
